jwt.go

```diff
@@ -101,6 +112,34 @@ func jwtMiddleware() gin.HandlerFunc {
 				return
 			}
 
+			keyParam := getRequestParam(c, "key")
+			bucketParam := getRequestParam(c, "bucket")
+
+			if keyParam == "" {
+				keyParam = getRequestParam(c, "prefix")
+			}
+
```

Comment:
Regarding REST API uploads https://github.com/MarshallWace/cachenator/blob/master/s3.go#L74, maybe this should be changed to `prefix` but can be done in another PR as it's a breaking change.

Add an extra if for the upload case

```suggestion

if keyParam == "" {
	keyParam = getRequestParam(c, "path")
}

```